### PR TITLE
Closes #2 | Create dataset loader for XNLI 

### LINKED ISSUE #2: Create dataset loader for XNLI
Dataloader name: xnli
DataCatalogue: seacrowd.github.io/seacrowd-catalogue/card.html?xnli

|     Dataset        | xnli  |
|-------------|---|
| Description | XNLI is a subset of a few thousand examples from MNLI which has been translated into a 14 different languages (some low-ish resource). As with MNLI, the goal is to predict textual entailment (does sentence A imply/contradict/neither sentence B) and is a classification task (given two sentences, predict one of three labels). |
| Subsets | th, vi |
| Languages | tha, vie |
| License | Creative Commons Attribution Non Commercial 4.0 (cc-by-nc-4.0) |
| Homepage | https://github.com/facebookresearch/XNLI |
| HF URL | https://huggingface.co/datasets/xnli |
| Paper URL | https://aclanthology.org/D18-1269/ |

### PR DESCRIPTION
**Pull request by khelli07**
Closes #2 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

image

image

### CODE REVIEW

**Comment by sabilmakbar** [comment]
Hi @khelli07 , thanks for the implementing this dataloader.

To pass the test, you can use --subset_id args in the test script so that the config name will look as follows:
{subset_id}_{source/seacrowd}_{config_suffix}

**Comment by khelli07** [comment]
Should the subset_id be schema specific too?
Suppose I have `subsets = ["xnli.tha", "xnli.vie"]`, then the args I inputed was `--subset_id xnli.tha` not `--subset_id xnli.tha_source` because it will throw 
```ValueError: BuilderConfig 'xnli.tha_source_source' not found. Available: ['xnli.tha_source', 'xnli.vie_source', 'xnli.tha_seacrowd_pairs', 'xnli.vie_seacrowd_pairs']```

But if I only input the  `--subset_id xnli.tha` then the test case will run both schemas (which I think it's ok, but need to make sure).

**Comment by sabilmakbar** [comment]
> But if I only input the --subset_id xnli.tha then the test case will run both schemas (which I think it's ok, but need to make sure).

No need, this one is already correct

**Review by SamuelCahyawijaya — CHANGES_REQUESTED** [review]
Hi @khelli07 , thanks for contributing! Can you change the subset naming from `xnli.<lang>_*` into `xnli_lang_*`

**Comment by khelli07** [comment]
I see, is there any standardization on this?
I was referencing from [MLQA](https://github.com/SEACrowd/seacrowd-datahub/blob/c646fad209ef80b0f821c4c16b079e58552ef3a5/seacrowd/sea_datasets/mlqa/mlqa.py#L53), but turns out [XQuad](https://github.com/SEACrowd/seacrowd-datahub/blob/c646fad209ef80b0f821c4c16b079e58552ef3a5/seacrowd/sea_datasets/xquad/xquad.py#L48) also use the same convention (using dot). However, I also find another using underscore: [Oscar](https://github.com/SEACrowd/seacrowd-datahub/blob/c646fad209ef80b0f821c4c16b079e58552ef3a5/seacrowd/sea_datasets/oscar_2201/oscar_2201.py#L235). 

Shouldn't we have uniformity on this?

**Review by SamuelCahyawijaya — APPROVED** [review]
LGTM!!

File: seacrowd/sea_datasets/xnli/xnli.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+}
+
+_SUPPORTED_TASKS = [Tasks.TEXTUAL_ENTAILMENT]
+
+_SOURCE_VERSION = "1.1.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class XNLI(datasets.GeneratorBasedBuilder):
```
Comment (sabilmakbar) [inline seacrowd/sea_datasets/xnli/xnli.py:66]: add `Dataset` on Dataloader Class Name
```suggestion
class XNLIDataset(datasets.GeneratorBasedBuilder):
```

File: seacrowd/sea_datasets/xnli/xnli.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(
+        self, dl_manager: datasets.DownloadManager
+    ) -> List[datasets.SplitGenerator]:
```
Comment (sabilmakbar) [inline seacrowd/sea_datasets/xnli/xnli.py:123]: change on one-liner based on code-formatter
```suggestion
    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
```

File: seacrowd/sea_datasets/xnli/xnli.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+                    "filepath": test_data_path,
+                    "split": "test",
+                    "language": _LANGUAGE_MAPPER[lang],
+                },
+            ),
+        ]
+
+    def _generate_examples(
+        self, filepath: Path, split: str, language: str
+    ) -> Tuple[int, Dict]:
```
Comment (sabilmakbar) [inline seacrowd/sea_datasets/xnli/xnli.py:167]: change on one-liner based on code-formatter
```suggestion
    def _generate_examples(self, filepath: Path, split: str, language: str) -> Tuple[int, Dict]:
```

**Review by sabilmakbar — CHANGES_REQUESTED** [review]
init comment, on code-formatter and Class Name

**Comment by sabilmakbar** [comment]
The dataloader works as expected. Once the changes have been addressed, I'll approve this, too, @khelli07.

**Comment by khelli07** [comment]
OK, will take a look into this tomorrow

File: seacrowd/sea_datasets/xnli/xnli.py (hunk 3 above)
Comment (khelli07) [reply seacrowd/sea_datasets/xnli/xnli.py:167]: I used black for formatting, I might need to see somewhere in the documentation about the Seacrowd's standard

Comment (sabilmakbar) [reply seacrowd/sea_datasets/xnli/xnli.py:167]: it uses `black` and `flake8`. you can rune the makefile using this on base dir:

`make check_file=seacrowd/sea_datasets/<dataset_name>/<dataset_name>.py`

**Review by sabilmakbar — APPROVED** [review]
LGTM! Thanks a lot @khelli07 for implementing this dataloader. Merging this...

**Pull request merged**